toolchain/sem_ir/formatter.cpp

```diff
@@ -144,61 +158,68 @@ auto Formatter::IncludeChunkInOutput(size_t chunk) -> void {
   }
 }
 
-auto Formatter::OverlapsWithDumpSemIRRange(
-    InstId inst_id, llvm::ArrayRef<InstBlockId> body_block_ids) -> bool {
+auto Formatter::ShouldFormatEntity(InstId decl_id, bool is_definition_start)
+    -> bool {
+  if (!decl_id.has_value()) {
+    return true;
+  }
+  if (!should_format_entity_(decl_id)) {
+    return false;
+  }
+
   if (!sem_ir_->parse_tree().tokens().has_dump_sem_ir_ranges()) {
     return true;
   }
 
-  auto loc_id = sem_ir_->insts().GetCanonicalLocId(inst_id);
+  // When there are dump ranges, ignore imported instructions.
+  auto loc_id = sem_ir_->insts().GetCanonicalLocId(decl_id);
   if (loc_id.kind() != LocId::Kind::NodeId) {
```

Comment:
Is it possible for the location to refer to `NodeId::None`? I think that would cause problems in the array indexing below.

Reply:
`None` would map to `LocId::Kind::None`